demisto_sdk/commands/common/tools.py

```diff
@@ -2244,9 +2257,129 @@ def is_test_config_match(
     if test_playbook_id:
         return test_playbook_match
 
+    if script_id:
+        return scripts_match
+
     return False
 
 
+def is_content_item_dependent_in_conf(test_config, file_type) -> bool:
+    """Check if a line from conf have multiple integration/scripts dependent on the TPB.
+        For example the dict
+        {"integrations": ["PagerDuty v2", "PagerDuty v3"], "playbookID": "PagerDuty Test"}
+        for file type integration is dependent
+        and for file type playbook or testplaybook is not dependent.
+    Args:
+        test_config (int): The dict in the conf file.
+        file_type (str): The file type, can be integrations, scripts or playbook.
+
+    Returns:
+        bool: The return value. True for dependence, False otherwise.
+    """
+    integrations_from_dict = test_config.get("integrations", [])
+    integrations_list: list = (
+        integrations_from_dict
+        if isinstance(integrations_from_dict, list)
+        else [integrations_from_dict]
+    )
+    scripts_from_dict = test_config.get("scripts", [])
+    scripts_list: list = (
+        scripts_from_dict
+        if isinstance(scripts_from_dict, list)
+        else [scripts_from_dict]
+    )
+    if file_type == "integration":
+        if len(integrations_list) > 1:
+            return True
+        return False
+    if file_type == "script":
+        if len(scripts_list) > 1:
+            return True
+        return False
+    # if the file_type is playbook or testplaybook in the conf.json it does not dependent on any other content
+    elif file_type == "playbook":
+        return False
+    return True
+
+
+def search_and_delete_from_conf(
+    conf_json_tests: list,
+    content_item_id: str,
+    file_type: str,
+    test_playbooks: list,
+    no_test_playbooks_explicitly: bool,
+) -> List[dict]:
+    """Return all test section from conf.json file without the deprecated content item.
+
+    Args:
+        conf_json_tests (int): The dict in the conf file.
+        content_item_id (str): The  content item id.
+        file_type (str): The file type, can be integrations, scripts or playbook.
+        test_playbooks (list): A list of related test playbooks.
+        no_test_playbooks_explicitly (bool): True if there are related TPB, False otherwise.
+
+    Returns:
+        bool: The return value. True for dependence, False otherwise.
+    """
+    test_registered_in_conf_json = []
+    # If the file type we are deprecating is a integration - there are TBP related to the yml
+    if file_type == "integration":
```

Comment:
Too much duplicate code here. Let's do something like this:
```
if file_type in ["integration", "playbook"]:
    if file_type == "integration":
        content_item_id = "integration_id"
    else:  # file_type == "playbook"
        content_item_id = "test_playbook_id"

    test_registered_in_conf_json.extend(
        [
            test_config
            for test_config in conf_json_tests
            if is_test_config_match(test_config, **{keyword: content_item_id})
        ]
    )

```
For all use cases